logger.go

```diff
@@ -25,6 +25,10 @@ type ILogger interface {
 type ZapCompatibleLogger interface {
 	Desugar() *zap.Logger
 
+	// Not defined: Named(name string) *zap.SugaredLogger
```

Comment:
I had a whole episode of adding `Named` here before realizing why we explicitly need to avoid it.